api/src/test/java/org/openmrs/module/fgh/mpi/MpiHttpClientTest.java

```diff
@@ -1,11 +1,184 @@
+		PowerMockito.mockStatic(MpiContext.class);
+		PowerMockito.mockStatic(URL.class);
+		when(Context.getAdministrationService()).thenReturn(adminService);
+		when(adminService.getGlobalProperty(GP_AUTHENTICATION_TYPE)).thenReturn(AUTHENTICATION_TYPE.toString());
+		when(adminService.getGlobalProperty(GP_MPI_BASE_URL)).thenReturn(MPI_BASE_URL);
+		when(adminService.getGlobalProperty(GP_MPI_SYSTEM)).thenReturn(MPI_SYSTEM.toString());
+		when(adminService.getGlobalProperty(GP_MPI_APP_CONTENT_TYPE)).thenReturn(MPI_APP_CONTENT_TYPE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_FOCUS_REFERENCE)).thenReturn(MESSAGE_HEADER_REFERENCE);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_MESSAGE_HEADER_EVENT_URI)).thenReturn(MESSAGE_HEADER_EVENT_URI);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_ID)).thenReturn(SANTE_CLIENT_ID);
+		when(adminService.getGlobalProperty(GP_SANTE_CLIENT_SECRET)).thenReturn(SANTE_CLIENT_SECRET);
+		when(MpiUtils.getGlobalPropertyValue(GP_UUID_SYSTEM)).thenReturn(UUID_SYSTEM);
+		mpiContext.setAuthenticationType(AUTHENTICATION_TYPE.CERTIFICATE);
+		
+	}
+	
+	@Test
+	public void getPatient_shouldRetrievePatientInOpenCR() throws Exception {
+		
+		final String patientUuid = "patient-uuid-open-cr";
+		String baseUrl = "https://localhost:8080";
+		Map<String, Object> patientResponse = new HashMap<>();
+		patientResponse.put(FIELD_ACTIVE, true);
+		patientResponse.put(OPENMRS_UUID, patientUuid);
+		
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		when(mpiContext.getAuthenticationType()).thenReturn(AUTHENTICATION_TYPE.CERTIFICATE);
+		when(mpiContext.getServerBaseUrl()).thenReturn(baseUrl);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(patientResponse);
+		Map<String, Object> openCRResponse = mockMpiHttpClient.getPatient(patientUuid);
+		
+		assertEquals(openCRResponse.get(FIELD_ACTIVE), patientResponse.get(FIELD_ACTIVE));
+		assertEquals(openCRResponse.get(OPENMRS_UUID), patientUuid);
+	}
+	
+	@Test
+	public void getPatient_shouldRetrievePatientInSanteMpi() throws Exception {
+		final String patientUuid = "patient-uuid-sante-mpi";
+		String baseUrl = "https://localhost:8080";
+		Map<String, Object> patientResponse = new HashMap<>();
+		patientResponse.put(FIELD_ACTIVE, true);
+		patientResponse.put(OPENMRS_UUID, patientUuid);
+		
+		when(MpiContext.initIfNecessary()).thenReturn(mpiContext);
+		when(mpiContext.getAuthenticationType()).thenReturn(AUTHENTICATION_TYPE.OAUTH);
+		when(mockMpiHttpClient.getPatient(patientUuid)).thenReturn(patientResponse);
+		doNothing().when(mockMpiHttpClient).doAuthentication(any(String.class));
+		Map<String, Object> santeMPIResponse = mockMpiHttpClient.getPatient(patientUuid);
+		
+		assertEquals(santeMPIResponse.get(FIELD_ACTIVE), patientResponse.get(FIELD_ACTIVE));
+		assertEquals(santeMPIResponse.get(OPENMRS_UUID), patientUuid);
```

Comment:
I don't see where you are verifying that the patient request was submitted to SanteMPI

Reply:
Please, do extra research on how to write tests, you have to know the purpose of the logic you are testing and include in your test a verification check that indeed the logic under test did what it is supposed to do